fluss-server/src/main/java/com/alibaba/fluss/server/kv/KvManager.java

```diff
@@ -270,4 +288,41 @@ public KvTablet loadKv(File tabletDir) throws Exception {
         this.currentKvs.put(tableBucket, kvTablet);
         return kvTablet;
     }
+
+    private void dropRemoteKvSnapshot(PhysicalTablePath physicalTablePath, TableBucket tableBucket)
+            throws RemoteStorageException {
+        FsPath remoteKvTabletDir =
+                FlussPaths.remoteKvTabletDir(remoteKvDir, physicalTablePath, tableBucket);
+        FsPath remoteKvTableRootDir =
+                FlussPaths.remoteKvTableRootDir(remoteKvDir, physicalTablePath, tableBucket);
+        try {
+            if (remoteFileSystem.exists(remoteKvTabletDir)) {
+                remoteFileSystem.delete(remoteKvTabletDir, true);
+                LOG.info("Delete remote table bucket snapshot of {} success.", tableBucket);
+            }
+        } catch (Exception e) {
+            throw new RemoteStorageException(
+                    "Failed to delete remote kv tablet path:" + remoteKvTabletDir, e);
+        }
+        deleteEmptyParentRecursively(
+                remoteKvTabletDir.getParent(), remoteKvTableRootDir.getParent());
+    }
+
+    private void deleteEmptyParentRecursively(FsPath curDir, FsPath tableRootParentDir) {
```

Comment:
We shouldn't remove the table parent dir (i.e., the database dir), as it is just a table removal, not database removal. Even if the database is empty, we shouldn't remove it. 

We shouldn't remove the table dir in TabletServer, as it is only responsive for a single bucket. The removing of table dir should be done in Coordinator when all buckets are removed.

Reply:
Thank you for noticing this detail,it only delete bucket dir until table dir is empty and empty table dir will be deleted, the database dir won't be deleted even if it's empty.
i will update this code for deleting table dir by coordinate server, thinks for your guidance.

Reply:
> deleting table dir by coordinate server
Seems deleting table dir by coordinate server is missing in this pr? Right?